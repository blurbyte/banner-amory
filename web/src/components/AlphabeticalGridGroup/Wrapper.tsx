import styled from 'styled-components';

type WrapperProps = {
  columns: number;
};

const Wrapper = styled.div<WrapperProps>`
  display: grid;
  grid-column: span ${props => props.columns};
`;

export default Wrapper;
