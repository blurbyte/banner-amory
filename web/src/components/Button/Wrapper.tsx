import styled from 'styled-components';
import { transparentize } from 'polished';

import { Color } from '@styles/color';

const Wrapper = styled.button`
  position: relative;
  display: flex;
  justify-content: flex-end;
  width: 100%;
  height: 5rem;
  width: 15rem;
  background: ${transparentize(0.9, Color.Blue)};
  padding: 1.4rem 1rem 0 1rem;
  border: 0;
  outline: 0;
  border-top: 4px solid transparent;
  cursor: pointer;
  font-size: 1.8rem;
  font-weight: 700;
  user-select: none;
  transition: all 0.15s linear;

  &:focus {
    border-top: 4px solid ${Color.Blue};
  }
`;

export default Wrapper;
