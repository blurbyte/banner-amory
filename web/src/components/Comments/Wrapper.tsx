import styled from 'styled-components';

import { slideIn } from '@styles/animations';
import { Color } from '@styles/color';

const Wrapper = styled.div`
  margin-top: 5rem;
  opacity: 0;
  animation: ${slideIn()} 400ms ease-in forwards 1000ms;
  border-top: 1px solid ${Color.Blue};
`;

export default Wrapper;
