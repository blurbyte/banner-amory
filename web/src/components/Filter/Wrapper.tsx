import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  margin-right: 2rem;
  margin-top: 1rem;
`;

export default Wrapper;
