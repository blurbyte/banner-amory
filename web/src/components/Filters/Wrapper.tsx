import styled from 'styled-components';
import { transparentize } from 'polished';

import { Colors } from '@styles/colors';

const Wrapper = styled.div`
  background: ${transparentize(0.9, Colors.blue)};
`;

export default Wrapper;
