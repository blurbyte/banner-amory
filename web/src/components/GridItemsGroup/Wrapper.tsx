import styled, { keyframes } from 'styled-components';

const slideIn = keyframes`
  from {
    transform: translateY(-1.2rem);
    opacity: 0;
  }
  to {
    transform: translateY(0);
    opacity: 1;
  }
`;

type WrapperProps = {
  columns: number;
  animationDelay: number;
};

const Wrapper = styled.div<WrapperProps>`
  display: grid;
  grid-column: span ${props => props.columns};
  opacity: 0;
  animation: ${slideIn} 300ms ease-in forwards ${props => props.animationDelay}ms;
`;

export default Wrapper;
