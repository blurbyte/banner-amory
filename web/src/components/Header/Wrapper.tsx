import styled from 'styled-components';

import { Colors } from '@styles/colors';

const Wrapper = styled.header`
  height: 6.6rem;
  border-bottom: 1px solid ${Colors.blue};
`;

export default Wrapper;
