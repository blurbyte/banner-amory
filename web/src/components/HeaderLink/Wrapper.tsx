// wrapper is used here to avoid unknown props React warning

import styled from 'styled-components';
import { transparentize } from 'polished';

import { Color } from '@styles/color';
import Link from './Link';

type WrapperProps = {
  isActive?: boolean;
};

const Wrapper = styled.div<WrapperProps>`
  display: block;
  position: relative;
  z-index: 9999;
  width: 12rem;
  height: 5rem;
  background: ${props => (props.isActive ? Color.Red : transparentize(0.9, Color.Blue))};
  color: ${props => (props.isActive ? Color.Black : Color.Blue)};
  font-size: 1.8rem;
  font-weight: 700;
  text-decoration: none;
  margin-right: 3rem;
  user-select: none;
  transition: all 0.15s linear;

  & > ${Link}:focus {
    border-top: 4px solid ${props => (props.isActive ? Color.Black : Color.Blue)};
  }
`;

export default Wrapper;
