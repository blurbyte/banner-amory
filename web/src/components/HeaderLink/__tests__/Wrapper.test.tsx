import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Wrapper from '../Wrapper';

test('renders correctly', () => {
  const component = renderer.create(<Wrapper />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders correctly when isActive', () => {
  const component = renderer.create(<Wrapper isActive />);
  expect(component.toJSON()).toMatchSnapshot();
});
