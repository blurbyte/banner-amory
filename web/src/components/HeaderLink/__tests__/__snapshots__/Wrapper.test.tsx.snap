// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: block;
  position: relative;
  z-index: 9999;
  width: 12rem;
  height: 5rem;
  background: rgba(61,38,218,0.1);
  color: #3d26da;
  font-size: 1.8rem;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-right: 3rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
}

.c0 > .sc-bdVaJa:focus {
  border-top: 4px solid #3d26da;
}

<div
  className="c0"
/>
`;

exports[`renders correctly when isActive 1`] = `
.c0 {
  display: block;
  position: relative;
  z-index: 9999;
  width: 12rem;
  height: 5rem;
  background: #d92f29;
  color: #222222;
  font-size: 1.8rem;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-right: 3rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
}

.c0 > .sc-bdVaJa:focus {
  border-top: 4px solid #222222;
}

<div
  className="c0"
/>
`;
